### ISSUE TITLE
Distance is one too small if first string is longer

### ISSUE THREAD

**Issue by Cireo** [gfairchild/pyxDamerauLevenshtein#22]
I looked through the existing bugs and the documentation and didn't see anything pointing this out.  I was also unable to reproduce this with your reference python implementation [here](https://web.archive.org/web/20150909134357/http://mwh.geek.nz:80/2009/04/26/python-damerau-levenshtein-distance/)

```
$ python3 -m pip show pyxdameraulevenshtein
Name: pyxDamerauLevenshtein
Version: 1.6
Summary: pyxDamerauLevenshtein implements the Damerau-Levenshtein (DL) edit distance algorithm for Python in Cython for high performance.
Home-page: https://github.com/gfairchild/pyxDamerauLevenshtein
Author: Geoffrey Fairchild
Author-email: mail@gfairchild.com
License: BSD 3-Clause License
Location: /usr/local/lib/python3.7/site-packages
Requires: numpy
Required-by: 
```

```
>>> from pyxdameraulevenshtein import damerau_levenshtein_distance
>>> damerau_levenshtein_distance('1', '')                                                                                                                         
0
>>> damerau_levenshtein_distance('', '1')                                                                                                                         
1
```

**Comment by gfairchild** [comment]
Nice find. I'll try to get to this in a few days (work is insane right now unfortunately).

**Comment by Cireo** [comment]
No rush, I know the feeling.  Reordering seems to work fine.  One thing to maybe look into is a single https://hypothesis.readthedocs.io/en/latest/data.html?highlight=string#hypothesis.strategies.text test comparing the reference to your fast code, it probably would catch this and nearly anything else that someone could calculate by hand, and not take much effort

**Comment by svenski** [comment]
Added a patch and two tests to expose this bug, simple swap the variables if the second one is shorter than the first one -- avoiding getting into the implementation details. Feel free to reject if too hacky! 

Note, I didn't manage to compile the C file without loads of local path references.

**Comment by gfairchild** [comment]
This is really great, @svenski. I appreciate the PR. As you can see, I just haven't been able to get around to fixing this myself unfortunately. I'll review this shortly.

**Comment by gfairchild** [comment]
This is now fixed in 1.6.1, which is live on PyPI. Thanks a bunch, @svenski!
